app/src/main/java/unipd/se18/ocrcamera/AllergensManager.java

```diff
@@ -0,0 +1,254 @@
+package unipd.se18.ocrcamera;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import com.opencsv.CSVReader;
+
+import unipd.se18.ocrcamera.inci.Ingredient;
+
+
+
+/**
+ * Class that contains the methods to read the allergens from a file, create an allergen list,
+ * find the allergens in the recognized ingredients from the OCR.
+ */
+class AllergensManager {
```

Comment:
Have you considered to define an interface to implement before filling the class with methods? Because as for now it's not abstract at all while it certainly could be.